cat-launcher/src/pages/SettingsPage/hooks/useSettings.ts

```diff
@@ -0,0 +1,183 @@
+import { useEffect } from "react";
+import {
+  useMutation,
+  useQuery,
+  useQueryClient,
+} from "@tanstack/react-query";
+
+import { queryKeys } from "@/lib/queryKeys";
+import {
+  applySettings,
+  getSettings,
+  listFonts,
+  listThemes,
+} from "@/lib/commands";
+import { isThemeEqual } from "../lib/utils";
+import type { ThemeColors } from "@/generated-types/ThemeColors";
+
+export function useSettings(options?: {
+  onSettingsError?: (err: Error) => void;
+  onFontsError?: (err: Error) => void;
+  onThemesError?: (err: Error) => void;
+}) {
+  const {
+    data: settings,
+    isLoading: settingsLoading,
+    isError: settingsError,
+    error: settingsErrorObj,
+  } = useQuery({
+    queryKey: queryKeys.settings(),
+    queryFn: async () => {
+      return await getSettings();
+    },
+  });
+
+  const {
+    data: fonts,
+    isLoading: fontsLoading,
+    isError: fontsError,
+    error: fontsErrorObj,
+  } = useQuery({
+    queryKey: queryKeys.fonts(),
+    queryFn: async () => {
+      return await listFonts();
+    },
+  });
+
+  const {
+    data: themes,
+    isLoading: themesLoading,
+    isError: themesError,
+    error: themesErrorObj,
+  } = useQuery({
+    queryKey: queryKeys.themes(),
+    queryFn: async () => {
+      return await listThemes();
+    },
+  });
+
+  const isLoading = settingsLoading || fontsLoading || themesLoading;
+
+  useEffect(() => {
+    if (settingsError && options?.onSettingsError) {
+      options.onSettingsError(settingsErrorObj as Error);
+    }
+  }, [settingsError, settingsErrorObj, options?.onSettingsError]);
+
+  useEffect(() => {
+    if (fontsError && options?.onFontsError) {
+      options.onFontsError(fontsErrorObj as Error);
+    }
+  }, [fontsError, fontsErrorObj, options?.onFontsError]);
+
+  useEffect(() => {
+    if (themesError && options?.onThemesError) {
+      options.onThemesError(themesErrorObj as Error);
+    }
+  }, [themesError, themesErrorObj, options?.onThemesError]);
```

Comment:
In cat-launcher/src/pages/SettingsPage/hooks/useSettings.ts around lines 61 to
77, the three useEffect hooks include options?.onSettingsError (and similar) in
their dependency arrays which can cause infinite re-renders when callers pass
inline option objects; fix by capturing each callback into a stable ref (e.g.,
onSettingsErrorRef.current = options?.onSettingsError) and using the ref in the
effect body while removing the callback from the dependency array so the effects
only depend on the error values.